File: src/Azure/Mgmt/Models/Resource.cs
using System.Collections.Generic;

namespace Azure.Mgmt.Models {
    public class Resource {
        /// <summary>
        /// Gets resource Id
        /// </summary>
        public string Id { get; protected set; }

        /// <summary>
        /// Gets resource name
        /// </summary>
        public string Name { get; protected set; }

        /// <summary>
        /// Gets resource type
        /// </summary>
        public string Type { get; protected set; }

        /// <summary>
        /// Gets or sets resource location
        /// </summary>
        public string Location { get; set; }

        /// <summary>
        /// Gets or sets resource tags
        /// </summary>
        public IDictionary<string, string> Tags { get; set; }
    }
}

File: ruby/src/azure/mgmt/models/resource.rb
module Azure
  module Mgmt
    module Models
      class Resource
        attr_accessor :id, :name, :type, :location, :tags

        def initialize(name:, location:, tags:nil)
            self.id = "id_from_resource"
            self.name = name
            self.type = "type_from_resource"
            self.location = location
            self.tags = tags
        end
      end
    end
  end
end